.vib/nats/goss/goss.yaml

```diff
@@ -0,0 +1,28 @@
+file:
+  /bitnami/nats/conf/{{ .Vars.natsFilename }}.conf:
+    mode: "0644"
+    filetype: file
+    exists: true
+    owner: root
+command:
+  {{- $subject := printf "subj_%s" (randAlpha 5) }}
+  {{- $msg := printf "msg_%s" (randAlpha 5) }}
+  {{- $auth := printf "--user=%s --password=%s" .Vars.auth.user .Vars.auth.password }}
+  {{- $client_endpoint := printf "-s nats://nats:%d" .Vars.service.ports.client }}
+  manage-key:
+    ## We need to have the subscriber process running at the same time the publishing is done
+    ## The sleep is needed to avoid a race condition where the publishing is done before the subscriber is ready
+    ## The subscriber process needs to be killed so there is no dangling process making goss timeout
+    exec: ( USER=1001 nats {{ $auth }} {{ $client_endpoint }} sub {{ $subject }}; ) & sleep 3 && USER=1001 nats {{ $auth }} {{ $client_endpoint }} pub {{ $subject }} {{ $msg }} && sleep 3 && pkill -f 'nats '
+    ## Goss seems to regard the exit status of this test as unfinished (-1) even though the stdout is correct and the subscriber process is stopped
+    exit-status: -1
```

Comment:
I think you can avoid all of this if you use the `timeout` [command](https://man7.org/linux/man-pages/man1/timeout.1.html). You can try change it to something like this and see if it works:

```
$ (USER=1001 timeout 10 nats --user=test_nats_client --password='ComplicatedPassword123!4' sub -s nats://nats:4224 subj_wiggle || true) & &&
USER=1001 nats --user=test_nats_client --password='ComplicatedPassword123!4' pub -s nats://nats:4224 subj_wiggle here
15:40:06 Published 4 bytes to "subj_wiggle"
[#1] Received on "subj_wiggle"
here
```

Reply:
I do agree, it's smarter IMHO

Reply:
It is indeed, changed 👍